src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/AvoidSingleUseOfLocalJsonSerializerOptions.cs

```diff
@@ -0,0 +1,328 @@
+            if (operation is null)
+                return null;
+
+            while (operation.Parent is IConditionalOperation conditionalOperation)
+            {
+                operation = conditionalOperation;
+            }
+
+            return operation;
+        }
+
+        private static bool IsArgumentOfJsonSerializer(IOperation operation, INamedTypeSymbol jsonSerializerSymbol, out bool isArgumentOfInvocation)
+        {
+            if (operation.Parent is IArgumentOperation arg && arg.Parent is IInvocationOperation inv)
+            {
+                isArgumentOfInvocation = true;
+                return SymbolEqualityComparer.Default.Equals(inv.TargetMethod.ContainingType, jsonSerializerSymbol);
+            }
+
+            isArgumentOfInvocation = false;
+            return false;
+        }
+
+        private static bool IsFieldOrPropertyAssignment(ILocalReferenceOperation operation)
+        {
+            IOperation? current = operation.Parent;
+
+            while (current is IAssignmentOperation assignment)
+            {
+                if (assignment.Target is IFieldReferenceOperation or IPropertyReferenceOperation)
+                {
+                    return true;
+                }
+
+                current = current.Parent;
+            }
+
+            return false;
+        }
+
+        private static bool IsTupleForDeconstructionTargetingFieldOrProperty(IOperation operation)
+        {
+            IOperation? current = operation.Parent;
+
+            if (current is not ITupleOperation tuple)
+            {
+                return false;
+            }
+
+            Stack<int> depth = new Stack<int>();
+            depth.Push(tuple.Elements.IndexOf(operation));
+
+            // walk-up right-hand nested tuples.
+            while (tuple.Parent is ITupleOperation parent)
+            {
+                depth.Push(parent.Elements.IndexOf(tuple));
+                tuple = parent;
+            }
+
+            current = tuple.WalkUpConversion().Parent;
+            if (current is not IDeconstructionAssignmentOperation deconstruction)
+            {
+                return false;
+            }
+
+            // walk-down left-hand nested tuples and see if it targets a field or property.
+            if (deconstruction.Target is not ITupleOperation deconstructionTarget)
+            {
+                return false;
+            }
+
+            tuple = deconstructionTarget;
+
+            IOperation? target = null;
+            while (depth.Count > 0)
+            {
+                int idx = depth.Pop();
+                target = tuple.Elements[idx];
+
+                if (target is ITupleOperation targetAsTuple)
+                {
+                    tuple = targetAsTuple;
+                }
+                else if (depth.Count > 0)
+                {
+                    return false;
+                }
+            }
+
+            return target is IFieldReferenceOperation or IPropertyReferenceOperation;
```

Comment:
NIT: Lots if effort for checking if the tuple targeting field or property, did not see a test case that flags a usage with tuple, is there any case that should be flagged? isn't it better to just bail out if the option is used within an tuple?

Reply:
Yes, it is probably too much effort for a rare case. But it doesn't hurt to have it. I can add the missing test you are referring to.